src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts

```diff
@@ -0,0 +1,168 @@
+                // Close the connection
+                await hubConnection.stop();
+            }
+        });
+    });
+
+    it("stream invocation", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const connection = new TestConnection();
+
+            const hubConnection = createHubConnection(connection, logger);
+            try {
+                hubConnection.stream("target", 1);
+
+                // Verify the message is sent
+                expect(connection.sentData.length).toBe(1);
+                expect((connection.sentData[0] as string).length).toEqual(63);
+            } finally {
+                // Close the connection
+                await hubConnection.stop();
+            }
+        });
+    });
+
+    it("upload invocation", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const connection = new TestConnection();
+
+            const hubConnection = createHubConnection(connection, logger);
+            try {
+                // We don't actually care to wait for the invoke.
+                // tslint:disable-next-line:no-floating-promises
+                hubConnection.invoke("target", 1, new Subject())
+                    .catch((_) => { }); // Suppress exception and unhandled promise rejection warning.
+
+                // Verify the message is sent
+                expect(connection.sentData.length).toBe(1);
+                expect((connection.sentData[0] as string).length).toEqual(81);
+            } finally {
+                // Close the connection
+                await hubConnection.stop();
+            }
+        });
+    });
+
+    it("upload stream invocation", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const connection = new TestConnection();
+
+            const hubConnection = createHubConnection(connection, logger);
+            try {
+                hubConnection.stream("target", 1, new Subject());
+
+                // Verify the message is sent
+                expect(connection.sentData.length).toBe(1);
+                expect((connection.sentData[0] as string).length).toEqual(81);
+            } finally {
+                // Close the connection
+                await hubConnection.stop();
+            }
+        });
+    });
+
+    it("completion message", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const connection = new TestConnection();
+
+            const hubConnection = createHubConnection(connection, logger);
+            try {
+                const subject = new Subject();
+                hubConnection.stream("target", 1, subject);
+                subject.complete();
+
+                await delayUntil(1000, () => connection.sentData.length === 2);
+
+                // Verify the message is sent
+                expect(connection.sentData.length).toBe(2);
+                expect((connection.sentData[1] as string).length).toEqual(29);
```

Comment:
Can we verify the message type is what we expect before verifying the length?

Reply:
Done